File: src/main/java/org/gjt/jclasslib/structures/attributes/targettype/ParameterTargetInfo.java
/*
 This library is free software; you can redistribute it and/or
 modify it under the terms of the GNU General Public
 License as published by the Free Software Foundation; either
 version 2 of the license, or (at your option) any later version.
 */

package org.gjt.jclasslib.structures.attributes.targettype;

import org.gjt.jclasslib.structures.InvalidByteCodeException;

import java.io.DataInput;
import java.io.DataOutput;
import java.io.IOException;

/**
 * Target info for a <tt>TypeAnnotation</tt> structure with a parameter target.
 */
public class ParameterTargetInfo extends TargetInfo {

    private int typeParameterIndex;

    public int getTypeParameterIndex() {
        return typeParameterIndex;
    }

    public void setTypeParameterIndex(int typeParameterIndex) {
        this.typeParameterIndex = typeParameterIndex;
    }

    @Override
    public void read(DataInput in) throws InvalidByteCodeException, IOException {
        super.read(in);
        typeParameterIndex = in.readUnsignedByte();
    }

    @Override
    public void write(DataOutput out) throws InvalidByteCodeException, IOException {
        super.write(out);
        out.writeByte(typeParameterIndex);
    }

    @Override
    public int getLength() {
        return 1;
    }

    @Override
    public String getVerbose() {
        return "parameter index " + typeParameterIndex;
    }
}

File: modules/data/src/main/kotlin/org/gjt/jclasslib/structures/attributes/targettype/ParameterTargetInfo.kt
/*
 This library is free software; you can redistribute it and/or
 modify it under the terms of the GNU General Public
 License as published by the Free Software Foundation; either
 version 2 of the license, or (at your option) any later version.
 */

package org.gjt.jclasslib.structures.attributes.targettype

import org.gjt.jclasslib.structures.InvalidByteCodeException

import java.io.DataInput
import java.io.DataOutput
import java.io.IOException

/**
 * Target info for a TypeAnnotation structure with a parameter target.
 */
class ParameterTargetInfo : TargetInfo() {

    var typeParameterIndex: Int = 0

    @Throws(InvalidByteCodeException::class, IOException::class)
    override fun read(input: DataInput) {
        typeParameterIndex = input.readUnsignedByte()
    }

    @Throws(InvalidByteCodeException::class, IOException::class)
    override fun write(output: DataOutput) {
        output.writeByte(typeParameterIndex)
    }

    override val length: Int
        get() = 1

    override val verbose: String
        get() = "parameter index $typeParameterIndex"

    override val debugMessage: String
        get() = "ParameterTargetInfo with $verbose"
}
